src/schemas/utils.ts

```diff
@@ -77,6 +78,36 @@ const convertClinicalFileErrorToGql = (fileError: {
   };
 };
 
+const convertRegistrationDataToGql = (
+  programShortName: string,
+  data: {
+    registration: DeepReadonly<ActiveRegistration> | undefined;
+    errors?: RegistrationErrorData[];
+    batchErrors?: { message: string; batchNames: string[]; code: string }[];
+  },
+) => {
+  const registration = get(data, 'registration', {} as Partial<typeof data.registration>);
+  const schemaAndValidationErrors = get(data, 'errors', [] as typeof data.errors);
+  const fileErrors = get(data, 'batchErrors', [] as typeof data.batchErrors);
+  return {
+    id: registration?._id || undefined,
+    programShortName,
+    creator: registration?.creator || undefined,
+    fileName: registration?.batchName || undefined,
+    createdAt: registration?.createdAt || undefined,
```

Comment:
optional chaining short circuits to `undefined` already

previous to this being available to us calls would throw an error

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

Reply:
Got my git steps out of order, legacy syntax fixes will be included in [#1051 1006](https://github.com/icgc-argo/argo-clinical/pull/1051/files)